### PR TITLE
New socket plugin

### PR DESCRIPTION
**Pull request by niicojs**
### Short Description:

New socket plugin to use with the online UI
### Changes:
- Add new socket plugin
- Some new events
- Fix object JSON serialisation (for long id)

@OpenPoGo/maintainers

### CODE REVIEW

**Comment by MaxLeiter** [comment]
Please update the README plugins list

File: api/item.py [hunk 1]
```diff
@@ -10,3 +10,16 @@ def __init__(self, data):
...
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state["unique_id"] = str(self.unique_id)
+        state["pokemon_id"] = str(self.pokemon_id)
+        return state
+
+    def __setstate__(self, state):
+        state["unique_id"] = int(state["unique_id"])
+        state["pokemon_id"] = int(state["pokemon_id"])
+        self.__dict__.update(state)
```
Comment (wchill) [inline api/item.py]: if `JSONEncodable` works for this these should be removed now

**Comment by andythorne** [comment]
FYI, I'm half way through a refactor of the evolve plugin to be able to evolve a specific pokemon :)

**Comment by wchill** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/OpenPoGo/OpenPoGoBot/pull-request/232/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=OpenPoGo/OpenPoGoBot)

**Comment by MaxLeiter** [comment]
:+1:  LGTM

[Approved with PullApprove](https://pullapprove.com/OpenPoGo/OpenPoGoBot/pull-request/232/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=OpenPoGo/OpenPoGoBot)

**Pull request merged**